TeamCode/src/main/java/org/firstinspires/ftc/teamcode/modules/driveTrain/DriveTrain.kt

```diff
@@ -20,52 +31,111 @@ object DriveTrain : IRobotModule {
     private val _velocityPidfSide = PIDRegulator(Configs.DriveTrainConfig.VELOCITY_PIDF_SIDE)
     private val _velocityPidfRotate = PIDRegulator(Configs.DriveTrainConfig.VELOCITY_PIDF_ROTATE)
 
-    override fun init(collector: BaseCollector) {
+    private var _isAuto: Boolean = false
+
+    private lateinit var _eventBus: EventBus
+
+    private lateinit var _battery: Battery
+
+    override fun init(collector: BaseCollector, bus: EventBus) {
+        _eventBus = bus
+
+        _battery = collector.devices.battery
+        _isAuto = collector.isAuto
+
         _leftForwardDrive = collector.devices.leftForwardDrive
         _rightForwardDrive = collector.devices.rightForwardDrive
         _leftBackDrive = collector.devices.leftBackDrive
         _rightBackDrive = collector.devices.rightBackDrive
 
-        _rightBackDrive.direction = DcMotorSimple.Direction.REVERSE
-        _rightForwardDrive.direction = DcMotorSimple.Direction.REVERSE
-    }
+        _leftForwardDrive.zeroPowerBehavior = DcMotor.ZeroPowerBehavior.BRAKE
+        _rightForwardDrive.zeroPowerBehavior = DcMotor.ZeroPowerBehavior.BRAKE
+        _leftBackDrive.zeroPowerBehavior = DcMotor.ZeroPowerBehavior.BRAKE
+        _rightBackDrive.zeroPowerBehavior = DcMotor.ZeroPowerBehavior.BRAKE
 
-    private fun driveSimpleDirection(direction: Vec2, rotate: Double) {
-        _leftForwardDrive.power = direction.x - direction.y - rotate
-        _rightBackDrive.power = direction.x - direction.y + rotate
-        _leftBackDrive.power = direction.x + direction.y - rotate
-        _rightForwardDrive.power = direction.x + direction.y + rotate
-    }
+        _leftBackDrive.direction = DcMotorSimple.Direction.REVERSE
+        _leftForwardDrive.direction = DcMotorSimple.Direction.REVERSE
 
-    private var _targetDirectionVelocity = Vec2.ZERO
-    private var _targetRotateVelocity = 0.0
+        bus.subscribe(SetDrivePowerEvent::class){
+            var dir = it.direction
+            var rot = it.rotate
+
+            if(_eventBus.invoke(IntakeManager.RequestLiftPosEvent()).pos != IntakeManager.LiftPosition.TRANSPORT) {
+                dir *= Vec2(Configs.DriveTrainConfig.LIFT_MAX_SPEED)
+                rot *= Configs.DriveTrainConfig.LIFT_MAX_ROTATE_SPEED
```

Comment:
В явном виде указать размерность этих переменных - это не скорость, а **процент** на который мы её домножаем. Например:
```suggestion
                dir *= Vec2(Configs.DriveTrainConfig.LIFT_MAX_SPEED_PCNT)
                rot *= Configs.DriveTrainConfig.LIFT_MAX_ROTATE_SPEED_PCNT
```